### ISSUE TITLE
Path to H3 install

### ISSUE THREAD

**Issue by psyhlo** [vcmi/vcmi#1870]
**Describe your proposal**
I think it will be good to let user choose if they want to copy H3 files or set a path to where is installed already.

**Additional context**
it is to reduce duplicating data on your drive. Or as i'm on windows dont want to copy data files in Windows drive and want to have several versions of VCMI installed for testing. Release, Dev and beta.

May be it would be good to separate mods for different builds? Something like Mods\Release Mods\Develop Mods\Beta and so on

**Comment by kambala-decapitator** [comment]
You could also use symbolic links / junctions as an alternative

**Comment by kambala-decapitator** [comment]
Btw looks like a duplicate of #874

**Comment by psyhlo** [comment]
> You could also use symbolic links / junctions as an alternative

yeah i could do that but it is not user friendly

**Comment by dydzio0614** [comment]
doesn't new launcher welcome screen copy h3 files to location where all VCMI instances read from?

**Comment by psyhlo** [comment]
yes but that just duplicate the files and i dont want that
and if it is made as i propose it would allow you to have several different installations of VCMI with different mods and the same H3 game data files

**Comment by dydzio0614** [comment]
considering current priorities i'd say this task is good for year 2035 :grin:

**Comment by IvanSavenko** [comment]
> yeah i could do that but it is not user friendly

To be fair, in my opinion current approach (data copy) is good enough.
Sure, it leads to data duplication, but it also relatively fool-proof - if player were to move or delete his H3 data vcmi will keep using its copied data as before.

And if player want to avoid duplication and share data he can setup symbolic links.

No objections against such an option, but personally see no need for it.

> and if it is made as i propose it would allow you to have several different installations of VCMI with different mods and the same H3 game data files

Somewhat different issue.
If you want to de-duplicate data you would want to have:
1) shared H3 data path (as proposed in this issue)
2) different user data path (since this is where Launcher manages mods)

(already possible on Linux by the way - since we use environment variables to detect paths)

**Comment by psyhlo** [comment]
> Somewhat different issue. If you want to de-duplicate data you would want to have:
> 
> 1. shared H3 data path (as proposed in this issue)
> 2. different user data path (since this is where Launcher manages mods)
> 
> (already possible on Linux by the way - since we use environment variables to detect paths)

it would be good if the user have option to choose where to install launcher data (copy H3 files. mods and settings), because in my case i dont have much space in my Windows drive and would like to move it elsewhere. without making symlinks and so on

**Comment by krs0** [comment]
Same complaint from my side. VCMI on windows takes from precious SSD space.

**Comment by dydzio0614** [comment]
I am kinda surprised that 0.5 GB of data is big deal in 2023 though

**Comment by IvanSavenko** [comment]
> I am kinda surprised that 0.5 GB of data is big deal in 2023 though

Yeah... I think I have at least 5 various H3 copies on my 1 TB SSD's

**Comment by krs0** [comment]
128GB here.... I hope solution would not be to upgrade my pc because of... H3 :)
Half joking.... anyway, let the user decide what to install where. To change this is not blocked by a real constraint from VCMI's side.

**Comment by IvanSavenko** [comment]
- Implemented in #5611
